src/tls13.c

```diff
@@ -6870,13 +6870,15 @@ int DoTls13Finished(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
         ret = BuildTls13HandshakeHmac(ssl, secret, mac, &finishedSz);
     #if defined(OPENSSL_ALL) || defined(WOLFSSL_HAPROXY) || defined(WOLFSSL_WPAS)
         if (ssl->options.side == WOLFSSL_CLIENT_END) {
-            XMEMCPY(ssl->serverFinished, mac, finishedSz);
-            ssl->serverFinished_len = finishedSz;
-        }
-        else {
-            XMEMCPY(ssl->clientFinished, mac, finishedSz);
-            ssl->clientFinished_len = finishedSz;
+            if (finishedSz <= TLS_FINISHED_SZ_MAX) {
+                XMEMCPY(ssl->serverFinished, mac, finishedSz);
+                ssl->serverFinished_len = finishedSz;
+            }
         }
+        else if (finishedSz <= TLS_FINISHED_SZ_MAX) {
```

Comment:
Prefer check of finishSz around outside check of side.
Why would this happen? We are building the MAC and we should know the maximum size.

Reply:
Agree. This is redundant. Removed the check